src/sentry/integrations/github/client.py

```diff
@@ -331,24 +330,15 @@ def should_count_api_error(self, error: ApiError, extra: dict[str, str]) -> bool
 
         return should_count_error
 
-    def get_repos(self, fetch_max_pages: bool = False) -> list[dict[str, Any]]:
+    def get_repos(self) -> list[dict[str, Any]]:
         """
-        args:
-         * fetch_max_pages - fetch as many repos as possible using pagination (slow)
-
         This fetches all repositories accessible to the Github App
         https://docs.github.com/en/rest/apps/installations#list-repositories-accessible-to-the-app-installation
 
         It uses page_size from the base class to specify how many items per page.
         The upper bound of requests is controlled with self.page_number_limit to prevent infinite requests.
         """
-        if not fetch_max_pages:
-            fetch_max_pages = options.get("github-app.fetch-max-pages")
-        return self.get_with_pagination(
-            "/installation/repositories",
-            response_key="repositories",
-            page_number_limit=self.page_number_limit if fetch_max_pages else 1,
```

Comment:
`self.page_number_limit` is the default, thus, we don't need to pass it anymore.